### DISCUSSION TITLE
Convert YAML definitions to PHP

### DISCUSSION THREAD

**Discussion by ruudk** [overblog/GraphQLBundle#889] [Q&A] [answered]
Does somebody have experience with converting YAML definitions to PHP classes?

I want to do this in an automated way as we have hundreds of them.

**Comment by mcg-web** [comment, marked as answer]
Yaml types are always converted to php classes for a performance purpose. You can use command graphql: compile to generate them on demand.

**Reply by ruudk** [reply]
Does that mean that I can just take the generated PHP files and use those as my definition source? Or do you mean I can take those as a reference point?

When I define my definition in PHP, will the bundle still generate these types on the fly? Or does it not do that when the definition is already in PHP?

**Reply by murtukov** [reply]
First of all, what do you mean by "Defining types with PHP"? Do you mean PHP Annotations (feature provided by the bundle) or PHP Types (feature provided by the underlying library `webonyx/graphql-php`)?

**Reply by ruudk** [reply]
I mean PHP Types:
```php
final class EventCategory extends EnumType implements AliasedInterface
 // ..
}
```

**Reply by murtukov** [reply]
@ruudk this bundle is just a wrapper for `webonyx/graphql-php` and it basically just generates PHP Types for you from YAML, Annotations or GraphQL SDL. The generated classes are Symfony services. That means that you can just take the generated PHP classes and use them as your source (as you assumed). Well, technically these classes are already used as your sources, so the only thing you have to do is just to move them from the output folder somewhere alse and make them services and that's it. The GraphQLBundle will use these classes automatically. Here are some examples: https://github.com/overblog/GraphQLBundle/issues/683

**Reply by ruudk** [reply]
@murtukov Thanks, I tried it and copied all the compiled PHP files into my `src` directory, but I still notice that the bundle takes all them and compiles them again (same output, except namespace) to my defined `cache_dir`. Why does it have to do that? I just want to reference all my code from the `src` directory, without compilation phase.

**Reply by ruudk** [reply]
I think I solved it:
```php
                'use_classloader_listener' => false,
                'auto_compile'             => false,
                'show_debug_info'          => '%kernel.debug%',
                'config_validation'        => false,
                'mappings'                 => [
                    'auto_discover' => [
                        'root_dir' => false,
                        'bundles'  => false,
                        'built_in' => false,
                    ],
                ],
```

**Reply by murtukov** [reply]
@ruudk when you are in dev environment it always recompiles your php types from YAML, Annotations etc. In prod mode it does it only once, until you clear the cache.

The `'auto_compile' => false` option disables the autocompilation completely. You don't have to set any other options to disable it.

**Reply by ruudk** [reply]
Thanks for the help. I was able to convert all my YAML configs to PHP and disable auto_compile. 🚀
